credential.go

```diff
@@ -254,7 +294,29 @@ func (d *DisclosureProofBuilder) Commit(randomizers map[string]*big.Int) []*big.
 		}
 		list = append(list, l...)
 	}
-	return list
+
+	if d.rpStructures != nil {
+		d.rpCommits = make(map[int][]*rangeproof.ProofCommit)
+		// we need guaranteed order on index
+		for index := 0; index < len(d.attributes); index++ {
+			structures, ok := d.rpStructures[index]
+			if !ok {
+				continue
+			}
+			g := rangeproof.NewQrGroup(d.pk.N, d.pk.R[index], d.pk.S)
+			d.rpCommits[index] = nil
```

Comment:
This too seems to be a no-op.

Reply:
fixed

recently worked too much with languages with maps that dont return null on non-set keys but just blow up in your face.